### ISSUE TITLE
codegen eqs with deps as functions, per offline design discussion

### ISSUE THREAD

**Issue by jzacsh** [rxie25/PLT2016Spring#86]
**tl;dr** @tiancizhong and i discussed/sketched how to codegen an **eqs' deps** as of _(see #56)_

... not sure if everyone else is up to date. whoever takes on this issue _(or is interested in it)_, if you're not sure what to do, ping here and i'll fill this in with a better description.

At the least, fixing this bug requires getting tests skipped in 3699e0c7a031c60 to pass:
- [ ] **low hanging fruit**: `tests/test-find-decl-assignment.eq`

  is actually just broken codegen test case i discovered while working on equation dependencies. totally unrelated to equation deps.
- [ ] `tests/test-lazyresolved-vars-increment-self.eq`
- [ ] `tests/test-lazyresolved-vars.eq`
- [ ] `tests/test-global-variable-overwrite.eq`
- [ ] `tests/test-global-variable`

**Comment by jzacsh** [comment]
@tiancizhong i think you might be the best qualified to point to what's wrong with `tests/test-find-decl-assignment.eq` _(unrelated to equation dependencies - just a bug in our existing codegen - **before** pull #56)_

**Comment by jzacsh** [comment]
Hey @nnhoang how's it going? Need any help?

**Comment by jzacsh** [comment]
nam's pull request #91 is nearly complete. just needs to fix an issue with `range` statements that was recently merged from master. after he merges, we'll have **only 1(!!) skipped test remaining**, and it is [only because we don't feel it is important](https://github.com/rxie25/PLT2016Spring/blob/master/src/tests/test-find-noargs.skip.eq)
